### PR TITLE
KB-2925 | New API  for the OTP generation for email/phone verification

### PR DESCRIPTION
**Pull request by tarentomaheshvakkund**
1. OTP access token enhancements

### CODE REVIEW

File: service/src/main/java/org/sunbird/actor/otp/OTPActor.java [hunk 1]
```diff
@@ -396,11 +397,13 @@ private void verifyOTPV3(Request request) {
               request.getRequestContext(),
               "OTP_VALIDATION_SUCCESS:OTPActor:verifyOTP: Verified successfully Key = "
                       + OTPUtil.maskId(key, type));
-      int length = 16;
       Map<String, Object> parametersMap = new HashMap<>();
       parametersMap.put(JsonKey.TYPE, type);
       parametersMap.put(JsonKey.KEY, key);
-      String contextToken = generateRandomString(length);
+      Map<String,String> contextDetailsMap =  new HashMap<>();
+      contextDetailsMap.put(JsonKey.CONTEXT_TYPE,(String) otpDetails.get(JsonKey.CONTEXT_TYPE.toLowerCase()));
```
Comment (karthik-tarento) [inline service/src/main/java/org/sunbird/actor/otp/OTPActor.java]: toLowerCase should not be used.
Add the conversion for the column name in cassandratablecolumn.properties file

File: core/platform-common/src/main/resources/externalresource.properties [hunk 2]
```diff
@@ -116,4 +116,6 @@ user_read_api_v2_non_mandatory_fields= profileDetails.verifiedKarmayogi,profileD
   profileDetails.professionalDetails[0].designation,profileDetails.professionalDetails[0].location,\
   profileDetails.professionalDetails[0].doj,profileDetails.professionalDetails[0].description,profileDetails.employmentDetails.departmentName
 kafka_user_first_login_event_topic= dev.user.first.login
+otpValidationSecretKey=secretKey
+otpExpirationTimeToken=300000
```
Comment (karthik-tarento) [inline core/platform-common/src/main/resources/externalresource.properties:120]: change this value to 5 mins

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: service/src/main/java/org/sunbird/actor/otp/OTPActor.java (hunk 1 above)
Comment (tarentomaheshvakkund) [reply service/src/main/java/org/sunbird/actor/otp/OTPActor.java]: 1. Added the entries to the cassandratablecolumn.properties

File: core/platform-common/src/main/resources/externalresource.properties (hunk 2 above)
Comment (tarentomaheshvakkund) [reply core/platform-common/src/main/resources/externalresource.properties:120]: its for 5mins only karthik i check on google for value.

**Pull request merged**